### PR TITLE
[DEX] Trade View Scaffold

### PR DESCRIPTION
**Pull request by schaeferCode**
## Ticket
* None

## Changes
* Transfer most of the code from wallet to mobile regarding Trade view

## Screenshots
screen shot 2018-09-20 at 7 58 14 pm

## Notes
I'm not using live data. There's still a lot that needs to be done to tie it together. 

UPDATE: I just added more stuff for the footer.

### CODE REVIEW

File: src/assets/scss/_globals.scss [hunk 1]
```diff
@@ -9,3 +9,7 @@ $font-size-low-res: 13px;
 @import 'fonts';
 @import 'tiles';
 @import 'type';
+
+%selected-text {
+  color: #000000;
+}
```
Comment (schaeferCode) [inline src/assets/scss/_globals.scss]: Not sure where this came from. I didn't make this change...

File: src/components/dex/Trade.vue [hunk 2]
```diff
@@ -0,0 +1,523 @@
...
+      return 'ATI'; // this.$t('amountQuote', { quote: this.currentMarket.quoteCurrency });
+    },
+
+    aphBalanceToolTip() {
+      return 'walletBalanceContractBalance'; // TODO: Fix this
+    },
+
+    aphHolding() {
+      if (this.currentMarket && this.$store.state.holdings) {
+        const holding = _.find(this.$store.state.holdings, (o) => {
```
Comment (crishellco) [inline src/components/dex/Trade.vue]:
```
const holding = _.find(this.$store.state.holdings, {assetId: this.$services.assets.APH});
```

File: src/components/dex/Trade.vue [hunk 3]
```diff
@@ -0,0 +1,523 @@
...
+      };
+    },
+
+    baseBalanceToolTip() {
+      return 'walletBalanceContractBalance'; // TODO: Fix this
+    },
+
+    baseHolding() {
+      if (this.currentMarket && this.$store.state.holdings) {
+        const holding = _.find(this.$store.state.holdings, (o) => {
```
Comment (crishellco) [inline src/components/dex/Trade.vue]:
```
const holding = _.find(this.$store.state.holdings, {assetId: this.currentMarket.baseAssetId});
```

File: src/components/dex/Trade.vue [hunk 4]
```diff
@@ -0,0 +1,523 @@
...
+    },
+
+    currentMarket() {
+      return this.$store.state.currentMarket;
+    },
+
+    estimate() {
+      const holding = this.currentMarket ?
+        this.$services.neo.getHolding(this.currentMarket.baseAssetId).unitValue :
+        0;
```
Comment (crishellco) [inline src/components/dex/Trade.vue:159]: Why `0` if the ternary evals to false?

File: src/components/dex/Trade.vue [hunk 5]
```diff
@@ -0,0 +1,523 @@
...
+        symbol: this.currentMarket ? this.currentMarket.baseCurrency : '',
+        balance: 0,
+        totalBalance: 0,
+        contractBalance: 0,
+      };
+    },
+
+    canPlaceMarketOrder() {
+      const currentWallet = this.$services.wallets.getCurrentWallet();
+      return currentWallet && currentWallet.isLedger !== true;
```
Comment (crishellco) [inline src/components/dex/Trade.vue]: `return currentWallet && !currentWallet.isLedger;`

File: src/components/dex/Trade.vue [hunk 6]
```diff
@@ -0,0 +1,523 @@
...
+        return new BigNumber(this.total).multipliedBy(
+          new BigNumber(holding));
+      } catch (e) {
+        console.log(e);
+        return 0;
+      }
+    },
+
+    isMarketClosed() {
+      return this.$store.state.currentMarket && this.$store.state.currentMarket.isOpen === false;
```
Comment (crishellco) [inline src/components/dex/Trade.vue]: `return this.$store.state.currentMarket && !this.$store.state.currentMarket.isOpen`

File: src/components/dex/Trade.vue [hunk 7]
```diff
@@ -0,0 +1,523 @@
...
+      }
+      return price;
+    },
+
+    priceLabel() {
+      if (!this.currentMarket) {
+        return '';
+      }
+
+      return 'GAS'; // this.$t('priceBase', { base: this.currentMarket.baseCurrency });
```
Comment (crishellco) [inline src/components/dex/Trade.vue]: `return this.currentMarket ? 'GAS' : '';`

File: src/components/dex/Trade.vue [hunk 8]
```diff
@@ -0,0 +1,523 @@
...
+    },
+
+    quoteBalanceToolTip() {
+      return 'walletBalanceContractBalance'; // TODO: Fix this
+    },
+
+    quoteHolding() {
+      if (this.currentMarket && this.$store.state.holdings) {
+        const holding = _.find(this.$store.state.holdings, (o) => {
```
Comment (crishellco) [inline src/components/dex/Trade.vue]:
```
const holding = _.find(this.$store.state.holdings, {assetId: this.currentMarket. quoteAssetId});
```

File: src/components/dex/Trade.vue [hunk 9]
```diff
@@ -0,0 +1,523 @@
...
+      }
+      if (this.orderType === 'Market') {
+        return !this.$store.state.orderQuantity || this.$isPending('placeOrder');
+      }
+      return !this.$store.state.orderQuantity || !this.$store.state.orderPrice || this.$isPending('placeOrder');
+    },
+
+    total() {
+      try {
+        if (!this.$store.state.orderQuantity) {
```
Comment (crishellco) [inline src/components/dex/Trade.vue]: This can be before the try/catch.

File: src/components/dex/Trade.vue [hunk 10]
```diff
@@ -0,0 +1,523 @@
...
+    };
+  },
+
+  methods: {
+    confirmOrder() {
+      // TODO: Fill this in with API request
+    },
+
+    loadHoldings() {
+      this.$store.dispatch('fetchHoldings', { done: null });
```
Comment (crishellco) [inline src/components/dex/Trade.vue]: No need for `done: null`.

File: src/components/dex/Trade.vue [hunk 11]
```diff
@@ -0,0 +1,523 @@
...
+    },
+
+    loadHoldingsSilently() {
+      this.$store.dispatch('fetchHoldings', { done: null, isRequestSilent: true });
```
Comment (crishellco) [inline src/components/dex/Trade.vue]: No need for `done: null`.

File: src/components/dex/Trade.vue [hunk 12]
```diff
@@ -0,0 +1,523 @@
...
+        }
+        quantityRemaining = quantityRemaining.minus(takeQuantity);
+        totalMultiple = totalMultiple.plus(takeQuantity.multipliedBy(level.price));
+      });
+
+      return (totalMultiple / quantity).toString();
+    },
+
+    percent(value) {
+      if (this.side === 'Buy') {
```
Comment (crishellco) [inline src/components/dex/Trade.vue]: `return this.side === 'Buy' ? this.percentForBuy(value) : this.percentForSell(value);`

File: src/services/dex.js [hunk 13]
```diff
@@ -24,4 +26,60 @@ export default {
...
     });
   },
+
+  formOrderBook(asks, bids) {
+    const book = {
+      asks: [],
+      bids: [],
+    };
+
+    asks.forEach((l) => {
```
Comment (crishellco) [inline src/services/dex.js]: `l`?

File: src/services/dex.js [hunk 14]
```diff
@@ -24,4 +26,60 @@ export default {
...
+      l.price = toBigNumber(l[0]);
+      l.quantity = toBigNumber(l[1]);
+      book.asks.push(l);
+    });
+    bids.forEach((l) => {
```
Comment (crishellco) [inline src/services/dex.js]: `l`?

File: src/services/dex.js [hunk 15]
```diff
@@ -24,4 +26,60 @@ export default {
...
+      totalBid = totalBid.plus(l.quantity);
+    });
+
+    if (book.asks.length > 0 && book.bids.length > 0) {
+      book.spread = book.asks[0].price - book.bids[0].price;
+      book.spreadPercentage = Math.round((book.spread / book.asks[0].price) * 10000) / 100;
+    }
+
+    let runningAsks = new BigNumber(0);
+    book.asks.forEach((l) => {
```
Comment (crishellco) [inline src/services/dex.js]: `l`?

File: src/services/dex.js [hunk 16]
```diff
@@ -24,4 +26,60 @@ export default {
...
+      runningAsks = runningAsks.plus(l.quantity);
+      l.quantityTotalRatio = runningAsks.dividedBy(totalAsk);
+      l.quantityRatio = l.quantity.dividedBy(totalAsk);
+    });
+    let runningBids = new BigNumber(0);
+    book.bids.forEach((l) => {
```
Comment (crishellco) [inline src/services/dex.js]: `l`?

File: src/services/neo.js [hunk 17]
```diff
@@ -556,6 +562,29 @@ export default {
     });
   },
 
+  getHolding(assetId) {
+    const holding = _.find(store.state.holdings, (o) => {
```
Comment (crishellco) [inline src/services/neo.js]: `o`?

**Review by crishellco — CHANGES_REQUESTED** [review]
A few stylistic/coding standard changes, but otherwise it's good.

File: src/components/dex/Trade.vue (hunk 4 above)
Comment (schaeferCode) [reply src/components/dex/Trade.vue:159]: I'm honestly not sure. That's just how this particular computed method was written in the wallet.  I think I'd be able to answer this better when we're working with live data or more robust fake data.

File: src/services/dex.js (hunk 13 above)
Comment (schaeferCode) [reply src/services/dex.js]: HAHA! Ah dang it! I just copied this code over. I'll at it and come up with a more sensible name.

Comment (schaeferCode) [reply src/services/dex.js]: Okay I fixed them.

**Review by crishellco — APPROVED** [review]

**Pull request merged**